nova/gce/driver.py

```diff
@@ -0,0 +1,807 @@
+              block_device_info=None):
+        """Create a new instance/VM/domain on the virtualization platform.
+        Once this successfully completes, the instance should be
+        running (power_state.RUNNING).
+
+        If this fails, any partial instance should be completely
+        cleaned up, and the virtualization platform should be in the state
+        that it was before this call began.
+
+        :param context: security context <Not Yet Implemented>
+        :param instance: nova.objects.instance.Instance
+                         This function should use the data there to guide
+                         the creation of the new instance.
+        :param image_meta: image object returned by nova.image.glance that
+                           defines the image from which to boot this instance
+        :param injected_files: User files to inject into instance.
+        :param admin_password: set in instance. <Not Yet Implemented>
+        :param network_info:
+           :py:meth:`~nova.network.manager.NetworkManager.get_instance_nw_info`
+        :param block_device_info: Information about block devices to be
+                                  attached to the instance.
+        """
+        """
+        Steps to spwan instances:
+        1. Get GCE image info
+        2. Get appropriate network info
+        3. Instance security groups
+        4. Add ssh keys if required
+
+        """
+        compute, project, zone = self.gce_svc, self.gce_project, self.gce_zone
+        instance_name = instance['name']
+        operation = gceutils.create_instance(compute, project, zone,
+                                             instance_name)
+        gceutils.wait_for_operation(compute, project, zone, operation)
+        gce_instance = gceutils.get_instance(compute, project, zone,
+                                             instance_name)
+        # Update gce_id in openstack instance['metadata']
+        instance['metadata'].update({'gce_id': gce_instance['name']})
+        # TODO: Fix
+        instance['metadata'].update({'public_ip_address': '104.197.111.111'})
+        operation = gceutils.set_instance_metadata(
+            compute,
+            project,
+            zone,
+            gce_instance, [
+                {
+                    'key': 'openstack_id',
+                    'value': instance['uuid']
+                },
+            ],
+            operation='add')
+        gceutils.wait_for_operation(compute, project, zone, operation)
+        self._uuid_to_gce_instance[instance.uuid] = gceutils.get_instancce(
+            compute, project, zone, instance_name)
+
+    def snapshot(self, context, instance, image_id, update_task_state):
+        """Snapshot an image of the specified instance
+        :param context: security context
+        :param instance: nova.objects.instance.Instance
+        :param image_id: Reference to a pre-created image holding the snapshot.
+        """
+        pass
```

Comment:
Snapshot operation in OpenStack creates an image from the instance. Is this not supported by GCE?